net_fetcher: Missing Content-Length causes ArgumentError

### Description

net_fetcher fails to download when Content-Length header is missing. This seems to happen a lot when downloading releases from github repos. Unfortunately the problem is hard to replicate, as it seems the github servers do some caching or something. The first time you request a resource it is missing the Content-Length header, but subsequent requests seem to contain it. I'm able to replicate the failure several times a day though.

### Omnibus Version

5.2.0

### Platform Version

CentOS 5/6/7

### Replication Case

Use a github release archive as as source. Such as: https://github.com/pypa/setuptools/archive/7.0.tar.gz

### Build Output

```
[NetFetcher: setuptools] I | Downloading from `https://github.com/pypa/setuptools/archive/7.0.tar.gz'


Progress: |==========================/builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/fetchers/net_fetcher.rb:185:in `each': comparison of NilClass with 1363 failed (ArgumentError)
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/fetchers/net_fetcher.rb:185:in `min'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/fetchers/net_fetcher.rb:185:in `block in download'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:331:in `call'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:331:in `block (3 levels) in open_http'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/protocol.rb:411:in `call_block'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/protocol.rb:402:in `<<'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/protocol.rb:102:in `read'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http/response.rb:319:in `read_chunked'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http/response.rb:282:in `block in read_body_0'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http/response.rb:250:in `inflater'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http/response.rb:280:in `read_body_0'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http/response.rb:201:in `read_body'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:328:in `block (2 levels) in open_http'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http.rb:1423:in `block in transport_request'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http/response.rb:162:in `reading_body'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http.rb:1422:in `transport_request'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http.rb:1384:in `request'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:319:in `block in open_http'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/net/http.rb:853:in `start'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:313:in `open_http'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:724:in `buffer_open'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:210:in `block in open_loop'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:208:in `catch'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:208:in `open_loop'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:149:in `open_uri'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/core_extensions/open_uri.rb:51:in `open_uri'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:704:in `open'
	from /opt/omnibus-toolchain/embedded/lib/ruby/2.1.0/open-uri.rb:34:in `open'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/fetchers/net_fetcher.rb:189:in `download'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/fetchers/net_fetcher.rb:85:in `fetch'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/software.rb:785:in `fetch'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/project.rb:1060:in `block (3 levels) in download'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/thread_pool.rb:64:in `call'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/thread_pool.rb:64:in `block (4 levels) in initialize'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/thread_pool.rb:62:in `loop'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/thread_pool.rb:62:in `block (3 levels) in initialize'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/thread_pool.rb:61:in `catch'
	from /builds/omnibus-dms/vendor/bundle/ruby/2.1.0/gems/omnibus-5.2.0/lib/omnibus/thread_pool.rb:61:in `block (2 levels) in initialize'
Makefile:8: recipe for target 'build' failed
```

From my initial debugging OpenURI's `content_length_proc` returns `nil` when there is no Content-Length header. This results in `reported_total` being set to `nil`, which causes the following line to throw the exception.

```ruby
downloaded_amount = [step, reported_total].min
```

--------------------------------------------------
/cc @chef/omnibus-maintainers <- This ensures the Omnibus Maintainers team are notified to review this PR.

--- a/lib/omnibus/download_helpers.rb
+++ b/lib/omnibus/download_helpers.rb
@@ -72,7 +72,7 @@ module Omnibus
             progress_bar.total = total
           }
           options[:progress_proc] = ->(step) {
-            downloaded_amount = [step, reported_total].min
+            downloaded_amount = reported_total ? [step, reported_total].min : step
             progress_bar.progress = downloaded_amount
           }
         end

